docs/reference/engines.md

```diff
@@ -0,0 +1,104 @@
+# Engines
+
+## Available features
+
+Metashell supports different underlying compilers and preprocessors, but not all
+compilers support everything needed by Metashell. For example Boost.Wave can not
+evaluate template metaprograms, as it is a preprocessor, but it is still
+supported by Metashell for getting an insight into preprocessing.
+
+The functionality of Metashell has been split up into a number of features to
+make it easier to document what functionality is available for the different
+compilers:
+
+<!-- feature_info -->
+* <strong id="type_shell">type shell</strong><br />
+It is used for evaluating template metaprograms in Metashell.<br />
+<br />
+* <strong id="preprocessor_shell">preprocessor shell</strong><br />
+It is used for evaluating preprocessor metaprograms in Metashell in preprocessor mode.<br />
+<br />
+* <strong id="code_completer">code completer</strong><br />
+It is used for code-completion in the shell.<br />
+<br />
+* <strong id="header_discoverer">header discoverer</strong><br />
+It is used for analysing the included headers and the include path.<br />
+<br />
+* <strong id="metaprogram_tracer">metaprogram tracer</strong><br />
+It is used by MDB to get an insight into template instantiations.<br />
+<br />
+* <strong id="cpp_validator">cpp validator</strong><br />
+It is used for checking that the code entered into Metashell is valid C++ code.<br />
+<br />
+* <strong id="macro_discovery">macro discovery</strong><br />
+It is used for querying the defined macros.<br />
+<br />
+* <strong id="preprocessor_tracer">preprocessor tracer</strong><br />
```

Comment:
I've tested it with `mkdocs` (used by http://metashell.org), which worked fine. I don't know what readthedocs is using, but in case it is having issues, I'll fix it in another PR.